subsys/net/ip/net_pkt.c

```diff
@@ -1245,6 +1245,10 @@ struct net_buf *net_frag_read(struct net_buf *frag, u16_t offset,
 		goto error;
 	}
 
+	if (len == 0) {
```

Comment:
Doh', my bad. You are right, the len==0 check is not needed, sorry for the noise :(